<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>guiMainWindowClass</class>
 <widget class="QMainWindow" name="guiMainWindowClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>716</width>
    <height>557</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>EPROM Programmer</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QGroupBox" name="setupLabel">
        <property name="title">
         <string>Set the serial port and desired baud rate, then press 'Init' to link to programmer.</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Serial Port</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="serialPort">
             <property name="toolTip">
              <string>The serial port to use.</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Orientation::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Baud Rate</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="baudRate">
             <property name="toolTip">
              <string>The desired baud rate.</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Orientation::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="initButton">
             <property name="toolTip">
              <string>Initialise the serial link at the given baud rate.</string>
             </property>
             <property name="text">
              <string>Init</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QGroupBox" name="groupBox">
             <property name="title">
              <string>Flow Control</string>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <property name="leftMargin">
               <number>4</number>
              </property>
              <property name="topMargin">
               <number>4</number>
              </property>
              <property name="rightMargin">
               <number>4</number>
              </property>
              <property name="bottomMargin">
               <number>4</number>
              </property>
              <item>
               <widget class="QRadioButton" name="flowNone">
                <property name="text">
                 <string>None</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="flowRtsCts">
                <property name="text">
                 <string>RTS/CTS</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Orientation::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Timeout</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="timeOut">
             <property name="suffix">
              <string>s</string>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>1000</number>
             </property>
             <property name="value">
              <number>10</number>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_5">
             <property name="orientation">
              <enum>Qt::Orientation::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Device Type</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="deviceType">
             <property name="toolTip">
              <string>Set the type of the EPROM</string>
             </property>
             <item>
              <property name="text">
               <string>TMS2532</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>i2716</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>i2708</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>i8755</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QPushButton" name="loadHexFile">
          <property name="toolTip">
           <string>Load a hex file for write/verify.</string>
          </property>
          <property name="text">
           <string>Load hex file</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="saveHexFile">
          <property name="toolTip">
           <string>Save a hex file.</string>
          </property>
          <property name="text">
           <string>Save hex file</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_4">
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="checkButton">
          <property name="toolTip">
           <string>Check if the DUT is erased.</string>
          </property>
          <property name="text">
           <string>Check</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="readButton">
          <property name="toolTip">
           <string>Read data from the DUT.</string>
          </property>
          <property name="text">
           <string>Read</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="writeButton">
          <property name="toolTip">
           <string>Write data to the DUT.</string>
          </property>
          <property name="text">
           <string>Write</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="verifyButton">
          <property name="toolTip">
           <string>Verify the DUT data against loaded hex file.</string>
          </property>
          <property name="text">
           <string>Verify</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QScrollArea" name="scrollArea">
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <widget class="QWidget" name="scrollAreaWidgetContents">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>688</width>
           <height>262</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QTextEdit" name="textEdit"/>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>716</width>
     <height>24</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionOpen_HEX_file"/>
    <addaction name="actionSave_HEX_file"/>
    <addaction name="actionQuit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="windowTitle">
    <string>8755 Programmer</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionOpen_HEX_file">
   <property name="text">
    <string>Load HEX file...</string>
   </property>
  </action>
  <action name="actionSave_HEX_file">
   <property name="text">
    <string>Save HEX file...</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>readButton</tabstop>
  <tabstop>checkButton</tabstop>
  <tabstop>writeButton</tabstop>
  <tabstop>textEdit</tabstop>
  <tabstop>serialPort</tabstop>
  <tabstop>timeOut</tabstop>
  <tabstop>scrollArea</tabstop>
 </tabstops>
 <resources/>
 <connections/>
 <slots>
  <slot>openHexFileSlot(bool)</slot>
  <slot>saveHexFileSlot()</slot>
 </slots>
</ui>
